src/typescript/Scala.tmLanguage.ts

```diff
@@ -846,7 +838,11 @@ export const scalaTmLanguage: TmLanguage = {
           name: 'storage.modifier.access'
         },
         {
-          match: '\\b(synchronized|@volatile|abstract|final|lazy|sealed|implicit|opaque |override|@transient|@native)\\b',
+          match: '\\b(synchronized|@volatile|abstract|final|lazy|sealed|implicit|override|@transient|@native)\\b',
+          name: 'storage.modifier.other'
+        },
+        {
+          match: '(?<=^|\\s)\\b(transparent|opaque|infix|open|inline)\\b(?=[a-z\\s]*\\b(def|val|var|given|type|class|trait|object|enum)\\b)',
```

Comment:
Instead of allowing `[a-z\\s]*` between a soft keyword modifier and a keyword, I think we should aim to only highlight a chain of soft keywords followed by a keyword. With soft keywords, I think we should try to be as minimal as possible, only highlighting them as keywords if strictly necessary.

Instead of allowing `[a-z\\s]*`, perhaps the regex can match on multiple chained soft keywords in a row by doing something like `((transparent|opaque|infix|open|inline)\\b)+`? (might need some adjustment for allowing whitespace).

Reply:
Note that a soft modifier can be followed by a sequence of normal modifiers. `[a-z\\s]*` skips those modifiers to find the `def|val|...`.

Reply:
The alternative would be to list all modifiers separated by spaces instead of `[a-z\\s]*`